test/test_mps.py

```diff
@@ -1831,6 +1831,28 @@ def test_batch_norm_backward_weight_bias_gradients(self):
         self.assertEqual(bn_cpu.weight.grad, bn_mps.weight.grad, atol=1e-5, rtol=1e-5)
         self.assertEqual(bn_cpu.bias.grad, bn_mps.bias.grad, atol=1e-5, rtol=1e-5)
 
+    def test_batch_norm_mixed_dtypes(self):
+        # See issue: https://github.com/pytorch/pytorch/issues/154887#issue-3111485538
+        input_mps = torch.rand((2, 3, 4), dtype=torch.float16, device="mps")
```

Comment:
Looks like your change will enable any `dtype_a` -> `dtype_b` cast, not just float16 -> float32, so we should probably test all of the supported cases, using `@parametrize`.

Also, I think the CPU impl does not support all cases and raises an error instead, so MPS should probably error in the same cases